lib/rubygems/path_support.rb

```diff
@@ -77,4 +79,12 @@ def default_path
     end
     gem_path
   end
+
+  def expand(path)
+    if File.directory?(path)
+      File.realpath(path)
```

Comment:
not all systems have `File.realpath` implemented

Reply:
Any in particular?  We can probably consider @eregon well versed in most (if not all), as [spec tests on](https://github.com/ruby/spec):
```
MRI on 30 platforms and 4 versions
JRuby for both 1.7 and 9.x
TruffleRuby
Opal
```

I suspect almost all of the RubyCI.org test against it (spec).  Also, the spec file for [realpath](https://github.com/ruby/spec/blob/master/core/file/realpath_spec.rb) seems to only differentiate between windows and all else, and that may be out of date, as newer windows builds work correctly.  I know I tested it recently, but I can't remember where...

Reply:
Doesn't windows not support it?

Reply:
Older versions of windows did not support symlinks, so older versions of Ruby also did not (on Windows, although Ruby lagged behind).   Anyway, after a brief AFK, I remembered that I came across it in the PR to add a few tests that were skipped (on Windows) back in.  For the versions of Ruby currently tested against, realpath & symlinks work...

Reply:
@MSP-Greg Thank you for the praise :) I don't claim to know the availability of `File.realpath` offhand though.
I think it's only Ruby 1.8.7 which doesn't have `File.realpath`, does RubyGems still support such old Ruby versions?
Note that TravisCI & AppVeyor passed.